acceptance/tests/environment/negative/agent_run_should_fail_if_env_unreadable.rb

```diff
@@ -0,0 +1,57 @@
+test_name "C97899 - Agent run should fail if environment is unreadable" do
+
+  testdir = ''
+  env_path = ''
+  test_env = ''
+
+  step 'setup environments' do
+    testdir = create_tmpdir_for_user master, 'c97899_unreadable_envdir'
+    env_path = "#{testdir}/environments"
+    test_env = "#{env_path}/testing"
+
+    apply_manifest_on(master, <<-MANIFEST, :catch_failures => true)
+      File {
+        ensure => directory,
+        mode => "0770",
+        owner => #{master.puppet['user']},
+        group => #{master.puppet['group']},
+      }
+      file {
+        '#{env_path}':;
+        '#{env_path}/production':;
+        '#{test_env}':;
+        '#{test_env}/manifests':;
+        '#{test_env}/modules':;
+      }
+      file { '#{test_env}/manifests/site.pp':
+        ensure => file,
+        mode => "0640",
+        content => 'node default { notify { "Hello agent": } }',
+      }
+    MANIFEST
+  end
+
+  step 'change permissions of envdir to 644' do
+    on(master, "chmod 644 #{test_env}")
+  end
+
+  step 'verify environment fails with puppet agent run' do
+    master_opts = {
+      'main' => {
+        'environmentpath' => env_path,
+      }
+    }
+    with_puppet_running_on master, master_opts, testdir do
```

Comment:
The `master` in `with_puppet_running_on master` is actually the host with the `master` role defined in the beaker host configuration. It's confusing because beaker also supports methods like `on(host, puppet("master ..."))` which really does mean run the webrick master... 

Also, `ci:test:aio` always uses `puppetserver`, but `ci:test:git` uses `webrick`, and we no longer support or test passenger packages. Since the failure only occurs under `puppetserver`, I think that means the overall test will fail when run in `ci:test:git`, e.g. due to the `expect_failure` below? If that's true, then I think we should confine to `aio` only.